qmlti/lib.php

```diff
@@ -314,27 +333,30 @@ function init_db($db) {
         $sql = 'CREATE TABLE IF NOT EXISTS ' . TABLE_PREFIX . 'lti_outcome ' .
                '(result_sourcedid VARCHAR(255),' .
                ' score VARCHAR(255),' .
-               ' report_url VARCHAR(255),' .
                ' created DATETIME)';
         $ok = $db->exec($sql) !== FALSE;
       }
 
       if ($ok) {
         $sql = 'CREATE TABLE IF NOT EXISTS ' . TABLE_PREFIX .  LTI_Data_Connector::REPORTS_TABLE_NAME . ' ' .
-               '(context_id VARCHAR(255),' .
+               '(consumer_key VARCHAR(50) NOT NULL DEFAULT \'\',' .
+               ' context_id VARCHAR(255),' .
                ' assessment_id VARCHAR(255),' .
-               ' is_accessible TINYINT)';
+               ' is_accessible TINYINT,' .
+               'PRIMARY KEY (consumer_key, context_id, assessment_id))';
         $ok = $db->exec($sql) !== FALSE;
       }
 
       if ($ok) {
         $sql = 'CREATE TABLE IF NOT EXISTS ' . TABLE_PREFIX . LTI_Data_Connector::RESULTS_TABLE_NAME . ' ' .
-               '(context_id VARCHAR(255),' .
+               '(consumer_key VARCHAR(50) NOT NULL DEFAULT \'\',' .
+               ' context_id VARCHAR(255),' .
                ' assessment_id VARCHAR(255),' .
                ' customer_id VARCHAR(25),' .
                ' created DATETIME,' .
                ' score VARCHAR(255),' .
-               ' result_id VARCHAR(255))';
+               ' result_id INT,' .
+               'PRIMARY KEY (consumer_key, result_id))';
```

Comment:
expected userid to be part of the PK e.g select * from RESULTS_TABLE_NAME where consumerKey = "mycompany" AND context = "TBD" AND assessmentid = "Physics101" AND userid="Joe"